src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs

```diff
@@ -291,7 +307,12 @@ private BindingConfig GetDefaultBindingConfig()
         // Called by CheckResult.ApplyBinding to compute the binding. 
         internal (TexlBinding, ReadOnlySymbolTable) ComputeBinding(CheckResult result)
         {
-            var parse = result.ApplyParse();
+            ParseResult parseResult = null;
+
+            if (result.RootNode == null)
+            {
+                parseResult = result.ApplyParse();
+            }
```

Comment:
can we avoid this extra check. 
If result.RootNode is null, then ApplyParse() should nop. 

Essentially, setting a ParseResult on the CheckREsult "short circuits"  Parsing. 
But everything downstream doesn't care and we avoid extra checks like 312 and 336.